### ISSUE TITLE
Dummy argument with INTENT(INOUT) and ALLOCATABLE fails to be parsed

### ISSUE THREAD

**Issue by clementval** [omni-compiler/omni-compiler#202]
This code is valid FORTRAN but fails with the front-end. 

```fortran 
module mod1

contains

  subroutine sub1(array)
    integer, allocatable, intent(inout) :: array(:)

    allocate(array(10))
  end subroutine sub1

  subroutine sub2(array)
    integer, intent(in) :: array(:)

  end subroutine sub2

  subroutine sub3(array)
    integer, allocatable, intent(out) :: array(:)

    allocate(array(10))
  end subroutine sub3

end module mod1
```

```bash
"mod1.f90:mod1", line 6: warning: array has allocatable

"mod1.f90:mod1", line 8: compiler error: type attr error: symbol=array attribute=allocatable flags=0x00001002
```

**Comment by clementval** [comment]
Should be fixed by #203

**Comment by clementval** [comment]
Closed with #203
